share/eds/byzantine/bad_encoding.go

```diff
@@ -150,13 +150,16 @@ func (p *BadEncodingProof) Validate(header *header.ExtendedHeader) error {
 	if err != nil {
 		return err
 	}
-	rebuiltExtendedShares, err := codec.Encode(rebuiltShares[0 : len(shares)/2])
+
+	squareWidth := uint64(len(header.DAH.RowsRoots) / 2)
+
+	rebuiltExtendedShares, err := codec.Encode(rebuiltShares[0:squareWidth])
 	if err != nil {
 		return err
 	}
-	rebuiltShares = append(rebuiltShares, rebuiltExtendedShares...)
+	copy(rebuiltShares[squareWidth:], rebuiltExtendedShares)
```

Comment:
I'm not sure how this was working before, or why switching the codec broke this. this seems like what we're trying to do, but I could be wrong. If we don't do this, we end up with a slice that has a len == 6, which will break the nmt wrapper.

Reply:
Seems like the new implementation uses slices slightly differently.

@vgonkivs, pls review this change

Reply:
looks correct

Reply:
@evan-forbes, does this mean that the klauspost leopard was introduced in v0.10.0? I thought we migrated before, but we didn't have these issues.

Reply:
I forgot to switch the default code :sweat_smile: in v0.9.0, although we were using rsmt2d v0.7.0